swagger.yaml

```diff
@@ -0,0 +1,515 @@
+             - status
+             - customer
+             - user
+             - startDate
+             - endDate
+        - name: sort
+          required: false
+          description: |
+            sort billing accounts by status, name, startDate, endDate
+          in: query
+          type: string
+
+    post:
+      tags:
+        - billingaccount
+      security:
+        - Bearer: []
+      operationId: addBillingAccount
+      description: Create a new billing account, Admin user only.
+      parameters:
+        - in: body
+          name: body
+          required: true
+          schema:
+            $ref: '#/definitions/NewBillingAccountBodyParam'
+      responses:
+        '403':
+          description: No permission or wrong token
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        '201':
+          description: Returns the newly created billing account
+          schema:
+            $ref: "#/definitions/BillingAccountResponse"
+        '422':
+          description: Invalid input
+          schema:
+            $ref: "#/definitions/ErrorModel"
+
+  /billingaccounts/{billingAccountId}:
+    get:
+      tags:
+        - billingaccount
+      security:
+        - Bearer: []
+      description: Retrieve billing account by id, Admin user only.
+      responses:
+        '403':
+          description: No permission or wrong token
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        '404':
+          description: Not found
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        '200':
+          description: a billing account
+          schema:
+            $ref: "#/definitions/BillingAccountResponse"
+      parameters:
+        - $ref: "#/parameters/billingAcountIdParam"
+      operationId: getBillingAccount
+
+    patch:
+      tags:
+        - billingaccount
+      security:
+        - Bearer: []
+      operationId: updateBillingAccount
+      description: Update a billing account, Admin user only.
+      responses:
+        '403':
+          description: No permission or wrong token
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        '404':
+          description: Not found
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        '200':
+          description: Successfully updated billing account. Returns original and updated billing account object
+          schema:
+            $ref: "#/definitions/UpdateBillingAccountResponse"
+        '422':
+          description: Invalid input
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        default:
+          description: error payload
+          schema:
+            $ref: '#/definitions/ErrorModel'
+      parameters:
+        - $ref: "#/parameters/billingAcountIdParam"
+        - name: body
+          in: body
+          required: true
+          schema:
+            $ref: "#/definitions/BillingAccountBodyParam"
+
+  /billingaccounts/{billingAccountId}/users:
```

Comment:
@ajefts @jswheeler would it be simpler to replace this endpoint with 'eligibleUsers' array to the billing accounts object? To add / remove users that can use this billing account, one just adds/removes userIds from that array. A PUT / PATCH on the billingAccounts object can handle this.
@mess - Eventually though, we would want to associate billingAccounts to an organization. Anyone belonging to that organization would then have access to the billing account(s). We might need to have levels / groups within the org to associate specific billing accounts.